llvm/test/Transforms/ConstraintElimination/add-nuw.ll

```diff
@@ -454,3 +454,127 @@ if.end:                                           ; preds = %entry
 }
 
 declare void @use(i1)
+
+define i1 @add_nuw_neg_pr54224_i16(i16 %a) {
+; CHECK-LABEL: @add_nuw_neg_pr54224_i16(
+; CHECK-NEXT:  entry:
+; CHECK-NEXT:    [[NEG2:%.*]] = add nuw i16 [[A:%.*]], -305
+; CHECK-NEXT:    [[C_1:%.*]] = icmp ugt i16 0, [[NEG2]]
```

Comment:
I'm curious why this condition `0 > an unsigned number` is not considered `false`.

Reply:
> I'm curious why this condition 0 > an unsigned number is not considered false by the compiler.

Such trivial compares usually are eliminated by other passes before ConstraintElimination runs.

ConstraintElimination could also handle them, if we add a `>= 0` constraint for each unsigned condition we add. But. that's probably better as a separate change, not for `apple:stable/20211026`